app/javascript/mastodon/features/interaction_modal/index.jsx

```diff
@@ -18,62 +25,207 @@ const mapDispatchToProps = (dispatch) => ({
   },
 });
 
-class Copypaste extends React.PureComponent {
+const PERSISTENCE_KEY = 'mastodon_home';
+
+const isValidDomain = value => {
+  const url = new URL('https:///path');
+  url.hostname = value;
+  return url.hostname === value;
+};
+
+const valueToDomain = value => {
+  // If the user uses a URL to their profile page or server
+  if (/^https?:\/\//.test(value)) {
+    try {
+      return (new URL(value)).host;
+    } catch {
+      return '';
+    }
+  // If the user writes their full handle including username
+  } else if (value.includes('@')) {
+    const segments = value.split('@');
+
+    if (segments.length === 2) {
+      return segments[1];
+    } else {
+      return '';
+    }
+  }
+
+  return value;
+};
+
+class LoginForm extends React.PureComponent {
 
   static propTypes = {
-    value: PropTypes.string,
+    resourceUrl: PropTypes.string,
+    intl: PropTypes.object.isRequired,
   };
 
   state = {
-    copied: false,
+    value: localStorage ? (localStorage.getItem(PERSISTENCE_KEY) || '') : '',
+    expanded: false,
+    selectedOption: -1,
+    isLoading: false,
+    error: false,
+    options: [],
   };
 
   setRef = c => {
     this.input = c;
   };
 
-  handleInputClick = () => {
-    this.setState({ copied: false });
-    this.input.focus();
-    this.input.select();
-    this.input.setSelectionRange(0, this.input.value.length);
+  handleChange = ({ target }) => {
+    this.setState({ value: target.value, isLoading: true, error: false }, () => this._loadOptions());
+  };
+
+  handleSubmit = () => {
+    const { value } = this.state;
+    const { resourceUrl } = this.props;
+
+    const domain = valueToDomain(value);
+
+    if (!isValidDomain(domain)) {
+      this.setState({ error: true });
+      return;
+    }
+
+    if (localStorage) {
+      localStorage.setItem(PERSISTENCE_KEY, domain);
```

Comment:
Would it be worth making this an array value so I can have multiple domains remembered? E.g., I have accounts on Mastodon, Pixelfed, and Calckey, I'd wamt to pick between each here